numpy/fft/_pocketfft.py

```diff
@@ -79,13 +75,41 @@ def _raw_fft(a, n, axis, is_real, is_forward, inv_norm):
     return r
 
 
-def _unitary(norm):
-    if norm is None:
-        return False
-    if norm=="ortho":
-        return True
-    raise ValueError("Invalid norm value %s, should be None or \"ortho\"."
-                     % norm)
+def _go_forw(n, norm):
+    if n < 1:
+        raise ValueError(f"Invalid number of FFT data points ({n}) specified.")
+
+    if norm is None or norm == "backward":
+        return 1
+    elif norm == "ortho":
+        return sqrt(n)
+    elif norm == "forward":
+        return n
+    raise ValueError(f'Invalid norm value {norm}; should be "backward",\
+                     "ortho" or "forward".')
+
+
+def _go_back(n, norm):
+    if n < 1:
+        raise ValueError(f"Invalid number of FFT data points ({n}) specified.")
+
+    if norm is None or norm == "backward":
+        return n
+    elif norm == "ortho":
+        return sqrt(n)
+    elif norm == "forward":
+        return 1
+    raise ValueError(f'Invalid norm value {norm}; should be "backward",\
+                     "ortho" or "forward".')
+
+
+def _swap_direction(norm):
+    try:
+        return {"backward": "forward", None: "forward",
+                "ortho": "ortho", "forward": "backward"}[norm]
```

Comment:
Maybe this is micro-optimization, but I'd prefer a plain if-else over dict for simple cases like this...

Reply:
apart from how compact it is, I think this is superior to a plain if-else clause bcs it handles the `KeyError` exception very well; if it's alright with you I'd suggest we go ahead with the `try` clause

Reply:
What's wrong with this?
```python
if ...:
    # do this
elif ...:
    # do that
else:
    raise ValueError(...)
```

Reply:
If you move the dictionary into a module level constant, `dict` lookup can actually be faster on average than a series of if-else chains:

```python
_SWAP_DIRECTION_MAP = {...}

def _swap_direction(norm):
  try:
    return _SWAP_DIRECTION_MAP[norm]
  except KeyError:
    raise ValueError("")

```

There's not much in it though, a few 10s of nanoseconds at most I think.

Reply:
Thank you, Peter! Is it because this way we save time for the dict construction?

Like I said this is micro optimization, that I was just curious why dict was preferred is all.

Reply:
in my reply I didn't mean that using an `if` clause cannot handle the exception, I just meant that using a `try` clause makes it (IMO) cleaner and easier to read; other than that I'm happy with an `if` clause too;

initially I was thinking of writing it with an `if` clause, but @peterbell10 suggested the dict solution which I found much more streamlined compared to what I would have come up with

Reply:
i've added a commit moving the dictionary into a module level constant to accelerate the lookup; it looks like our two options are: (1) use a `try` clause with the dict lookup with the module level constant, (2) revert to an `if` clause for the direction swap instead; 

let me know what you think; from the comments I understand that these are comparable in terms of performance @peterbell10 @leofang

Reply:
>Is it because this way we save time for the dict construction?

Yes, if it's inside the function then the dict is reconstructed for every function call whereas the module level variable is constructed only once at import time.

As for dict vs if-else in a more general sense. I have no strong feeling either way, both are not that much code and both can be quite readable.